woocommerce/woocommerce
# [e2e] External sites - Include API tests for Pressable and WPCOM

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Updates to the current `e2e-pw` setup to run API tests against Pressable and WPCOM testing sites. 

Some API tests are skipped for these envs and the reasons for that are due to different environments. Tests are structured to work against `wp-env`, and here we have sites on Pressable and WPCOM, which are permanent and we're using them for every test run. 

I'm skipping failing tests for now, and we can address them one by one separately and fix/update them if possible. 



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. CI green - to be sure that changes in this PR are not affecting wp-env execution.
2. Locally - checkout this branch and run `pnpm test:e2e:with-env default-pressable tests/e2e-pw/tests/api-tests` and `pnpm test:e2e:with-env default-wpcom tests/e2e-pw/tests/api-tests`.  Make sure all of them are passing. 
3. Take a look at the code; does it make sense? A lot of changes are ESLint updates, but mainly it was adding `@skip-on-default-pressable` and `@skip-on-default-wpcom`



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Include API tests into test suites for Pressable and WPCOM. 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
